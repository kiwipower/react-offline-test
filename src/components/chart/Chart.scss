.Chart {
  margin-top: 30px;
}
