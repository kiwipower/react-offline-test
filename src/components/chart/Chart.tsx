import * as React from 'react';
import PieChart from 'react-minimal-pie-chart';

import './Chart.scss';
import {ProductionType} from "../../store/types";

interface Props {
    productionTypes: ProductionType[],
}

const COLORS = [
    '#9400D3',
    '#4B0082',
    '#000080',
    '#4B0082',
    '#0000FF',
    '#00FF00',
    '#FFFF00',
    '#FF7F00',
    '#FF0000',
];

export const Chart: React.FC<Props> =
    ({productionTypes}) => (
        <main className="Chart">
            <PieChart
                label={({ data, dataIndex }) =>
                    data[dataIndex].title
                }
                labelPosition={112}
                labelStyle={{
                    fontFamily: 'sans-serif',
                    fontSize: '5px'
                }}
                style={{
                    height: '300px'
                }}
                data={
                    productionTypes && productionTypes.map((productionType, index) => {
                        return {
                            title: productionType.fuel, value: productionType.perc, color: COLORS[index]
                        }
                    })
                }
            />
        </main>
    );
