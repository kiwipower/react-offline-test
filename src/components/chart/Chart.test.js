import React from 'react';
import { shallow } from 'enzyme';

import { Chart } from './Chart';

const productionTypes =  [
  { fuel: 'biomass', perc: 3.6 },
  { fuel: 'wind', perc: 6 },
  { fuel: 'solar', perc: 0.3 },
];

const defaultProps = { productionTypes };

describe('Chart', () => {
  test('should render correctly', () => {
    const component = shallow(<Chart {...defaultProps}/>);
    expect(component).toMatchSnapshot();
  });

  test('Should have a \\"Chart\\" class', () => {
    const wrapper = shallow(<Chart {...defaultProps}/>);
    expect(wrapper.hasClass('Chart')).toBe(true);
  });
});
