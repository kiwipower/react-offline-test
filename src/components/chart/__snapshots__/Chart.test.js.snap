// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart should render correctly 1`] = `
<main
  className="Chart"
>
  <ReactMinimalPieChart
    animate={false}
    animationDuration={500}
    animationEasing="ease-out"
    cx={50}
    cy={50}
    data={
      Array [
        Object {
          "color": "#9400D3",
          "title": "biomass",
          "value": 3.6,
        },
        Object {
          "color": "#4B0082",
          "title": "wind",
          "value": 6,
        },
        Object {
          "color": "#000080",
          "title": "solar",
          "value": 0.3,
        },
      ]
    }
    label={[Function]}
    labelPosition={112}
    labelStyle={
      Object {
        "fontFamily": "sans-serif",
        "fontSize": "5px",
      }
    }
    lengthAngle={360}
    lineWidth={100}
    paddingAngle={0}
    radius={50}
    ratio={1}
    rounded={false}
    startAngle={0}
    style={
      Object {
        "height": "300px",
      }
    }
  />
</main>
`;
